lib/collection/variable-scope.js

```diff
@@ -74,6 +74,14 @@ _.inherit((
         this.values = new VariableList(this, VariableList.isVariableList(values) ? values.toJSON() : values);
         // in above line, we clone the values if it is already a list. there is no point directly using the instance of
         // a variable list since one cannot be created with a parent reference to begin with.
+
+        _.assign(this, /** @lends VariableScope.prototype */ {
```

Comment:
single assignment. avoid lodash until more than a few are needed.

also - this is private and very low use-case as such an extra array does not need to be created for every variable scope. we can create it on first `.addLayer` call